.github/workflows-source/ci-main.yml

```diff
@@ -78,6 +79,11 @@ jobs:
       labels: dind-large
     steps:
       - <<: *checkout
+      # The git merge base has to be fetched such that bazel tests like did_git_test
+      # can compare HEAD against the merge base for compatibility.
+      - name: Fetch the git merge base
+        if: ${{github.event_name == 'pull_request'}}
+        run: git fetch origin "$CI_MERGE_REQUEST_TARGET_BRANCH_SHA"
```

Comment:
You might be able to get the same effect adding this to the checkout step:
```
        with:
          ref: ${{ github.head_ref }}
```

Reply:
Won't that just fetch the HEAD of the PR?

Reply:
Ah I think you can also add `fetch-depth: 256`

Reply:
Note that we already set fetch-depth to 256. The problem is that the merge base is very likely not an ancestor of the PR so it won't be fetched regardless of the depth.

Reply:
Hmm I see, if there's no other option we might have to do it like this. But I prefer to not add too much custom CI logic for just one test... Would it also work if we added `it fetch origin "$CI_MERGE_REQUEST_TARGET_BRANCH_SHA"` to `_did_git_test_impl` so it's only executed when that method is called?

Reply:
My thinking was that we'll likely have other tests as well which compare the PR against its merge base so they can piggy back on this fetching. 

Also it's usually a good smell to have side-effects like this in tests.

Reply:
Hmm I don't think we have any other tests that require the merge base as far as I recall. I also am a little unsure why it worked in GitLab without the additional fetch command, but is required now, but if this is the best practice we can keep it like this. 👍

Reply:
I also remember that I tried fetching in the test before but that failed when running locally since the .git directory is read-only in that case.

Reply:
What would be ideal if actions/checkout had a plural version of `ref` like:
```
with:
  refs:
    - ${{ github.head_ref }}
    - ${{ github.event.pull_request.base.sha }}
```
Let me see about requesting that as a feature.

Reply:
Yeah that would be nice! I tested it with `fetch-depth: 256` and that worked, but it might have just been lucky: https://github.com/dfinity/ic/pull/503/files

Reply:
BTW I'm testing with removing the fetching of the merge-base to see if it now magically works.

Reply:
The only thing I don't understand is why it failed for me before.

Reply:
Do you have the commit that failed?

Reply:
>BTW I'm testing with removing the fetching of the merge-base to see if it now magically works.
It works :)

Reply:
> Do you have the commit that failed?

I can't find it anymore. Possibly because I force-pushed over it on Friday. 

Maybe it works now because there's a slight difference between `github.event.pull_request.base.sha` and `git merge-base master HEAD`. Let's just see how this works in practise.